letsencrypt-auto-source/letsencrypt-auto.template

```diff
@@ -319,6 +319,32 @@ else
   BOOTSTRAP_VERSION="Unknown 1"
 fi
 
+PrevBootstrapVersion() {
+  if [ -f $VENV_BOOTSTRAP_VERSION ]; then
+    cat $VENV_BOOTSTRAP_VERSION
+  elif grep -Fqx "$BOOTSTRAP_VERSION" << "UNLIKELY_EOF"
+NO_BOOTSTRAP 1
+BootstrapDebCommon 1
```

Comment:
> One might be to say roughly if [ ! -f $VENV_BOOTSTRAP_VERSION ]; then echo "$BOOTSTRAP_VERSION" | awk '{print $1, 1}' > "$VENV_BOOTSTRAP_VERSION"; fi;. Then PrevBootstrapVersion could just unconditionally check the file.

I don't think this has the behavior I described above (unless we make a rule that all new bootstrap scripts start at version 2).

For example, my plan was to split Red Hat based systems where we're installing a newer Python from EPEL into a new bootstrap script rather than adding a conditional to the existing RH script. The reason for doing this is it minimizes the number of people who need to rebootstrap.

If we always write "BootstrapFunc 1" if no file exists, some Red Hat users that need to rebootstrap won't do so. `certbot-auto` won't have created the file, we'll see it's the first version of the script (despite it now being a different script) and assume there's nothing to do.

My plan was to never modify this list. The existing bootstrap scripts are special because they existed before we saved the bootstrap version.

> There might need to be changes to how NO_BOOTSTRAP is handled; you wouldn't want that getting written to a file. And you wouldn't want it to re-bootstrap unnecessarily if BOOTSTRAP_VERSION was "NO_BOOTSTRAP 1" (because --no-bootstrap was passed) and then control hit if [ -d "$VENV_PATH" -a "$BOOTSTRAP_VERSION" != "$(PrevBootstrapVersion)" ]; then.

That's a good point. I think `--no-bootstrap` should just sidestep this logic entirely. In fact, it might be a good idea to copy over the version from the last bootstrap script we did run for when the user (possibly accidentally) runs without `--no-bootstrap` in the future.